streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -52,22 +54,26 @@ public void setUp() {
     @Test
     public void testGetProducerConfigs() throws Exception {
         Map<String, Object> returnedProps = streamsConfig.getProducerConfigs("client");
-        assertEquals(returnedProps.get(ConsumerConfig.CLIENT_ID_CONFIG), "client-producer");
+        assertEquals(returnedProps.get(ProducerConfig.CLIENT_ID_CONFIG), "client-producer");
+        assertEquals(returnedProps.get(ProducerConfig.LINGER_MS_CONFIG), "100");
+        assertNull(returnedProps.get("DUMMY"));
     }
 
     @Test
     public void testGetConsumerConfigs() throws Exception {
         Map<String, Object> returnedProps = streamsConfig.getConsumerConfigs(null, "example-application", "client");
         assertEquals(returnedProps.get(ConsumerConfig.CLIENT_ID_CONFIG), "client-consumer");
         assertEquals(returnedProps.get(ConsumerConfig.GROUP_ID_CONFIG), "example-application");
-
+        assertEquals(returnedProps.get(ConsumerConfig.MAX_POLL_RECORDS_CONFIG), "1000");
+        assertNull(returnedProps.get("DUMMY"));
     }
 
     @Test
     public void testGetRestoreConsumerConfigs() throws Exception {
         Map<String, Object> returnedProps = streamsConfig.getRestoreConsumerConfigs("client");
         assertEquals(returnedProps.get(ConsumerConfig.CLIENT_ID_CONFIG), "client-restore-consumer");
         assertNull(returnedProps.get(ConsumerConfig.GROUP_ID_CONFIG));
+        assertNull(returnedProps.get("DUMMY"));
```

Comment:
Should probably check that the consumer and producer configs no longer have properties that are not defined in their config.